### PR TITLE
Makes Grills ZAP things when shit gets tossed at them.

### PR DESCRIPTION
**Pull request by Supermichael777**
50/50 for damage or zap. Should be about the same as actually touching the grill.
:cl:
tweak:Throwing objects at electrified grills is no longer totally safe. BZZZZZZT.
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
keep in mind tesla_zap doesn't bounce if the input power is below 1000 watts

File: code/game/objects/structures/grille.dm [hunk 1]
```diff
@@ -268,8 +268,16 @@
...
-		var/obj/item/I = AM
-		tforce = max(0, I.throwforce * 0.5)
+		if(prob(50))
+			var/obj/item/I = AM
+			tforce = max(0, I.throwforce * 0.5)
+		else
+			var/turf/T = get_turf(src)
+			var/obj/structure/cable/C = T.get_cable_node()
+			var/datum/powernet/R = C.powernet
+			P = R.avail
```
Comment (RemieRichards) [inline code/game/objects/structures/grille.dm]: what is P, there is no P here, this won't compile, P doesn't exist.

**Comment by Supermichael777** [comment]
the input power will never get that low unless the power is of due to this drawing from the mains.

File: code/game/objects/structures/grille.dm (hunk 1 above)
Comment (Supermichael777) [inline code/game/objects/structures/grille.dm]: ah shit oops

File: code/game/objects/structures/grille.dm [hunk 2]
```diff
@@ -268,8 +268,16 @@
...
+		if(prob(50))
+			var/obj/item/I = AM
+			tforce = max(0, I.throwforce * 0.5)
+		else
+			var/turf/T = get_turf(src)
+			var/obj/structure/cable/C = T.get_cable_node()
+			var/datum/powernet/R = C.powernet
+			var/P = R.avail
+			playsound(src.loc, 'sound/magic/LightningShock.ogg', 100, 1, extrarange = 5)
+			tesla_zap(src, 3, P * 0.08) //ZAP for 1/5000 of the amount of power, which is from 15-25 with 200000W
```
Comment (RemieRichards) [inline code/game/objects/structures/grille.dm]: why not ```R.avail * 0.08```
or one step further:
```C.powernet.avail * 0.08```

Comment (Supermichael777) [inline code/game/objects/structures/grille.dm]: oh yeah

**Comment by kevinz000** [comment]
How does this make sense..? I mean unless you're like RIGHT next to it..

File: code/game/objects/structures/grille.dm [hunk 3]
```diff
@@ -271,13 +271,13 @@
...
-		else
+		else if(anchored && !destroyed)
 			var/turf/T = get_turf(src)
 			var/obj/structure/cable/C = T.get_cable_node()
-			var/datum/powernet/R = C.powernet
-			var/P = R.avail
-			playsound(src.loc, 'sound/magic/LightningShock.ogg', 100, 1, extrarange = 5)
-			tesla_zap(src, 3, P * 0.08) //ZAP for 1/5000 of the amount of power, which is from 15-25 with 200000W
+			if(C)
+				var/P = C.powernet.avail * 0.08
```
Comment (RemieRichards) [inline code/game/objects/structures/grille.dm]: lol uh, no my point was that the P var isn't needed at all, just ```C.powernet.avail * 0.08```

**Comment by Supermichael777** [comment]
It doesn't need to make sense. it just needs to zap the grey-shirts untill they stop doing it.

**Comment by Cheridan** [comment]
wowwww how did this lightning strike me in the middle of this field

**Comment by Zilenan91** [comment]
I still think it's a bad idea, shocked grilles don't need this much of a buff. If you want them to be impassable just make them into walls.

**Comment by Lumipharon** [comment]
>wowwww how did this lightning strike me in the middle of this field

wow how did this mains strength power at best zap me from 3 meters away while standing inside a big meteal box that the grille is directly attached to and has to basically ignore to get to me.

This is literally pointless - all it means is people will have to stand more than 3 tiles away from a grille to break it unless you have gloves. Wew lad.

**Comment by Cheridan** [comment]
@Supermichael777 make it reach 7 tiles

**Comment by Iamgoofball** [comment]
ye, buff it to 7 tiles

**Comment by Zilenan91** [comment]
Why don't we merge this into the april fools branch and set the range to 90 tiles

**Comment by Lumipharon** [comment]
The memes never end.

**Comment by optimumtact** [comment]
bad idea I do not like it

**Comment by Alek2ander** [comment]
So this makes it highly dangerous or even impossible to escape from brig even if moronic security leaves you with tools.

**Comment by PKPenguin321** [comment]
>The memes never end.
 I'm going to enjoy throwing 500 pieces of paper in front of a broken window to space that's shocked, and watch it just instantly murder literally everyone that walks close enough to see it.

This right here
Is why we should merge this ASAP

**Comment by Zilenan91** [comment]
Actually yeah that sounds hilarious merge immediately

**Comment by Steelpoint** [comment]
This is amusing.

Speaking seriously though, electrified grilles are no threat at all to anyone who is not new to the game.

**Comment by Supermichael777** [comment]
I set it to three tiles to reduce the ability of people to weaponize it. Note that if your attempting to throw shit at it to proc it you will probably shock yourself with Tesla bounce the first time you hit someone, also that involves someone standing closer to the grille then you while your throwing at it and praying to rnjesus you proc it before they realize your acting weird. Now if a dude is standing opposite when you try to beak through....

**Comment by Shadowlight213** [comment]
Someone should try and see if they can somehow get the tesla zap to bounce and hit a brig apc. HONK.

**Comment by Lumipharon** [comment]
If the range is high/unlimited/whatever, then this is hugely abusable and makes shocked grilles at high power massive grief tools/unpassable walls to anyone without a gun (because bullets don't cause it shocks cuz lel).

if the range is short, it's literally pointless in 90% of cases, and the same as the above in situation where you can't stand back from the grille (like brig cells).

Either way this seems very poorly thought out.

**Comment by Zilenan91** [comment]
Yeah you can murder people with this in the brig, the grille shocks will be literally inescapable to people who are trapped in the cells.

**Comment by Iamgoofball** [comment]
Good thing we have admins.

**Comment by Lumipharon** [comment]
uwotm8?

>guy: admins pls help, got killed because of terrible mechanics changes
>admins: yeah ok I'll get right onto banning those grilles

**Comment by Iamgoofball** [comment]
Are you seriously so dense you don't know I'm talking about the people griefing with this or is this an ETHAN BRADBERRY prank

**Comment by Zilenan91** [comment]
It's not even necessarily griefing. If sec leaves anything on the floor accidentally, or if the prisoner uncuffs themselves their own cuffs will kill them if there's a breach outside.

**Comment by Iamgoofball** [comment]
breaches don't throw, they just move objects, which shouldn't proc this shit

**Comment by Zilenan91** [comment]
But bodies get shocked if they get pulled in by breaches, shouldn't that be standardized to both to be consistent?

**Comment by Iamgoofball** [comment]
bodies get shocked if they get pulled in by breaches because it's calling Bump()

the tesla_zap shit is only on hitby() which only procs on throwing and throwing related procs.

**Comment by Lumipharon** [comment]
@goof what are you talking about, grief has nothing to do with this.

Guy in brig cell, currently can smash the window and escape if sec aren't paying attention.
Now he'll just electrocute himself and die.
Alternatively, tator's target is in a brig cell, tator just throws shoes at the grille to roast the dude trapped inside.

**Comment by Cheridan** [comment]
>Alternatively, tator's target is in a brig cell, tator just throws shoes at the grille to roast the dude trapped inside.

oh god
someone cleverly using an environmental hazard on what is supposed to be a "metal death trap masquerading as a space station."

whatever have we come to

**Comment by PKPenguin321** [comment]
Yeah that sounds awesome to me

**Comment by Saegrimr** [comment]
GOOD PR

**Comment by Lumipharon** [comment]
>ignores everything else that I said and focuses on what is valid to merge the meme pr

Yeah ok bro.

**Comment by Zilenan91** [comment]
It's good :+1: 

Still allows people to break shocked grilles by standind farther away while also letting them create environmental hazards for others.

**Comment by CosmicScientist** [comment]
:-1: instant death.

Try an explosion that throws a lot of objects at the grill.

**Comment by PKPenguin321** [comment]
:+1: funniest way to die, died to it earlier and i couldnt stop laughing for like 5 minutes

**Comment by phil235** [comment]
:-1: This is a terrible change. Unbalanced, abusable and unrealistic. If there really is a majority of player who wants this, can we at least reduce the range to 1 tile?

**Comment by Dawgas** [comment]
There is a point when you don't listen to the players for the good of the game.

This would be one of those times, but it never seems to happen when its needed most - instead we get shitty wishes they were dank memes like this

**Comment by Iamgoofball** [comment]
@phil235 
how many times do we have to tell you realism doesn't matter when we have such unrealistic game mechanics

plus we can just say "the wires are S.M.R.T Anti-Tamper wires that shock people throwing shit at it

re: changing it to 1 tile: that makes it not stop people throwing shit at grilles to shock them

**Comment by Man-Shroom-** [comment]
reduce damage/range or add a chance of it not occuring? i dunno im no coder

**Comment by Shadowlight213** [comment]
We should just make it zap the thrower. Maybe make it work off of the last person to touch the object?
Only that person gets zapped.

**Comment by CosmicScientist** [comment]
@Shadowlight213 seekrit greytide nerf.

**Comment by phil235** [comment]
Welp, today I got zapped. I was 8 tiles away from the grille! Someone in the brig threw an item an exposed grille of the warden's office. I was completely outside the brig and yet the electric arc zapped the thrower, bounced off him and zapped me, putting me in orange health from full health. Is this really what we want???

**Comment by Iamgoofball** [comment]
i dont see the problem here.

**Comment by Lumipharon** [comment]
Simply ebin.

**Pull request merged**